packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx

```diff
@@ -8,12 +8,15 @@ import { ITERATOR_SYMBOL, withSetStateAllowed, sym } from 'enzyme/build/Utils';
 import './_helpers/setupAdapters';
 import { createClass } from './_helpers/react-compat';
 import { describeIf, itIf, itWithData, generateEmptyRenderData } from './_helpers';
-import { REACT013, REACT014, REACT16, is } from './_helpers/version';
+import { REACT013, REACT014, REACT15, REACT150_4, REACT16, is } from './_helpers/version';
 
 // The shallow renderer in react 16 does not yet support batched updates. When it does,
 // we should be able to go un-skip all of the tests that are skipped with this flag.
 const BATCHING = !REACT16;
 
+// some React versions pass undefined as an argument of setState callback.
+const CALLING_SETSTATE_CALLBACK_WITH_UNDEFINED = REACT15 && !REACT150_4;
```

Comment:
I guess this condition doesn't cover a case that React v15.4 calls setState callback with `undefined`.

But all tests are passing in my environment. The reason is `ifReact` is checking with React v16.2 because `enzyme-adapter-react-helper` has React v16.2 into its `node_modules`.
I'm not sure why `enzyme-adapter-react-helper` has React v16.2 into its `node_modules`.

Reply:
Thanks! I'll dig into it later.

Reply:
`enzyme-adapter-react-helper` shouldn't have `react` as a devDependencies?

https://github.com/airbnb/enzyme/blob/master/packages/enzyme-adapter-react-helper/package.json#L50

Reply:
since it's a peer dep, it also needs to be a dev dep.

Reply:
Other packages don't have react as a devDeps. Why does only this package have it?

Reply:
That's a bug leftover from the v3 migration; they all should - every peer dep in every npm package anywhere must always have the identical peer dep string as a dep or a dev dep.

Reply:
I've almost got a fix PR with a regression test; i'll put it up shortly.

Reply:
https://github.com/airbnb/enzyme/pull/1465